api.h

```diff
@@ -0,0 +1,23 @@
+#ifndef _API_H
+#define _API_H
+
+#include <stdio.h>
+
+typedef struct {
+    size_t argc;
+    char** argv;
+} arg_data;
+
+typedef struct string_t {
```

Comment:
types ending in `_t` are reserved for posix (c libraries). I'd avoid using that.

Reply:
Oh, didn't know that. I'll change that.